File: microservice-notification-v2/src/main/java/com/notification/application/exception/ThrowableMapper.java
package com.notification.application.exception;

import io.github.responsekit.core.StandardResponse;
import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.ext.ExceptionMapper;
import jakarta.ws.rs.ext.Provider;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Provider
public class ThrowableMapper implements ExceptionMapper<Throwable> {
    private static final Logger log = LoggerFactory.getLogger(ThrowableMapper.class);

    @Override
    public Response toResponse(Throwable exception) {
        log.error(exception.getMessage(), exception);

        return Response
                .status(500)
                .entity(StandardResponse.error().build())
                .build();
    }
}

File: microservice-pricing/src/main/kotlin/com/pricing/infra/web/exception/ThrowableMapper.kt
package com.pricing.infra.web.exception

import io.github.responsekit.core.StandardResponse
import jakarta.ws.rs.core.Response
import jakarta.ws.rs.ext.ExceptionMapper
import jakarta.ws.rs.ext.Provider
import org.slf4j.Logger
import org.slf4j.LoggerFactory


@Provider
class ThrowableMapper: ExceptionMapper<Throwable> {
    private val log: Logger = LoggerFactory.getLogger(ThrowableMapper::class.java)

    override fun toResponse(exception: Throwable): Response {
        log.error(exception.message, exception)

        return Response
            .status(Response.Status.INTERNAL_SERVER_ERROR)
            .entity(
                StandardResponse
                        .fail()
                        .build()
            )
            .build()
    }
}